File: wifidb/tools/utilites/swarm_gen_xml.php
<?php
define("SWITCH_SCREEN", "CLI");
define("SWITCH_EXTRAS", "export");

if(!(require('../daemon/config.inc.php'))){die("You need to create and configure your config.inc.php file in the [tools dir]/daemon/config.inc.php");}
$wdb_install = $daemon_config['wifidb_install'];
if($wdb_install == ""){die("You need to edit your daemon config file first in: [tools dir]/daemon/config.inc.php");}
require($wdb_install)."/lib/init.inc.php";

$header = '<?xml version="1.0"?>
<file_events>
';
$footer = "</file_events>";
$events = "";
$result = $dbcore->sql->conn->query("SELECT `points`,`title`,`username`,`date`,`aps` FROM `users_imports` order by `id` DESC");
#$array = $result->fetch_row();
#var_dump($array);
#die();
while($array = $result->fetch_row())
{
    $date = strtotime($array[3]);
    if($date == ""){continue;}
    switch($details)
    {
        case 1:
            $i = $array[4]+0;
            echo "User: ".$array[2]."\r\nDate: ".$date."\r\nTitle: ".$array[1]."\r\nAPs: ".$i."\r\n";
            while($i!=0)
            {
                #echo $i." ";
                $event = '<event date="'.$date.'000" filename="'.$array[1].$i.'" author="'.$array[2].'" />';
                $events .= $event."\r\n";
                echo ".";
                $i--;
            }
            break;
        default:
            $i = 0;
            $conn2 = new mysqli("192.168.1.18", "root", "saNsui20si", "wifi");
            $points_explode = explode("-", $array[0]);
            echo "User: ".$array[2]."\r\nDate: ".$array[3]."\r\nTitle: ".$array[1]."\r\n";
            foreach($points_explode as $key=>$point)
            {
                
                $point_exp = explode(",", $point);
                if(!@$point_exp[1])
                {
                    var_dump($point);
                    continue;
                }
                $id_exp = explode(":", $point_exp[1]);
                $id = $id_exp[0];

                $pointer_result = $conn2->query("SELECT `sectype` FROM `wifi0` WHERE `id`='$id'");
                $ap_array = $pointer_result->fetch_row();
                $pointer_result->free();
                switch ($ap_array[0])
                {
                    case "1":
                        $sec_type = "open";
                        break;
                    case "2":
                        $sec_type = "wep";
                        break;
                    case "3":
                        $sec_type = "sec";
                        break;
                    default:
                        $sec_type = "open";
                        break;
                }
                $event = '<event date="'.$date.'000" filename="'.$id.'.'.$sec_type.'" author="'.$array[2].'" />';
                $events .= $event."\r\n";
                #if($key != 0) echo "-";
                #echo $id;
                $i++;
            }
            echo "APs: ".$i;
            #die();
            break;
    }
    echo "\r\n\r\n------------------------------------------------------------------\r\n\r\n";
}
$xml_all = $header.$events.$footer;
file_put_contents("wifidb_imports.xml", utf8_encode($xml_all));
?>

File: wifidb/tools/utilites/swarm_gen_xml.py
__author__ = 'pferland'

import os, sys, cymysql, time, datetime
from Config import Config

daemon_cfg_obj = Config("/etc/wifidb/config.ini")
daemon_config = daemon_cfg_obj.ConfigMap("daemon")

www_config_obj = Config( daemon_config['wifidb_install']+"/lib/config.ini")
www_config = www_config_obj.ConfigMap("sql")

conn = cymysql.connect(host='172.16.1.18', user='root', passwd='saNsui20si', db='wifi')
#conn = cymysql.connect(host=www_config['sql_host'], user=www_config['db_user'], passwd=www_config['db_pwd'], db=www_config['db'])

header = "<?xml version=\"1.0\"?>\r\n<file_events>\r\n"
footer = "</file_events>"
events = ""

cur = conn.cursor()
cur.execute("SELECT `points`,`title`,`username`,`date`,`aps` FROM `wifi`.`users_imports` order by `id` DESC");

array = cur.fetchall()
total = len(array)
import_count = 0
total_aps = 0
for imports in array:
    if not imports[3]:
        continue
    date_pre = str(imports[3])
    date = int(time.mktime(datetime.datetime.strptime(date_pre, "%Y-%m-%d %H:%M:%S").timetuple()))
    import_count += 1
    i = 0
    print str(import_count) + "/" + str(total) + " : User: " + str(imports[2]) + " - Date: " + str(imports[3]) + " (" + str(date) + ")" + " - Title: " + str(imports[1]) + " - APcount: " + str(imports[4]);
    author = str(imports[2])
    points_explode = str(imports[0]).split("-")
    for point in points_explode:

        point_exp = point.split(",")
        gps_id = point_exp[0]

        id_new_old = point_exp[1].split(":")
        apid = id_new_old[0]
        new_old = id_new_old[1]

        cur.execute("SELECT `sectype` FROM `wifi0` WHERE `id`= %s" % apid)
        sectype = cur.fetchone()
        if sectype[0] == '1':
            sec_type = "open"

        if sectype[0] == '2':
            sec_type = "wep"

        if sectype[0] == '3':
            sec_type = "wpa"

        if not sec_type:
            sec_type = "open"
                                                                        #
        events += "\t" + '<event date="' + str(date) + '000" filename="' + str(apid) + '.' + str(sec_type) + '" author="' + str(author) + '" />' + "\r\n";
        i += 1

    print "APs: " + str(i)
    total_aps += i
    print total_aps
    print "------------------------------------------------------------------"


xml_all = header + events + footer

print "write events.xml to current directory"
f = open("events.xml", "w")
f.write(xml_all)
f.close()
